nixos/modules/services/misc/podgrab.nix

```diff
@@ -0,0 +1,50 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.podgrab;
+in
+{
+  options.services.podgrab = with lib; {
+    enable = mkEnableOption "Podgrab, a self-hosted podcast manager";
+
+    passwordFile = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "/run/secrets/password.env";
+      description = ''
+        The path to a file containing the PASSWORD environment variable
+        definition for Podgrab's authentification.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      example = 4242;
+      description = "The port on which Podgrab will listen for incoming HTTP traffic.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.podgrab = {
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p /var/lib/podgrab/
+        rm -f /var/lib/podgrab/{client,webassets}
+        ln -s ${pkgs.podgrab}/share/{client,webassets} /var/lib/podgrab/
```

Comment:
Sometimes you **must** do this because the software won't work without it. Those circumstances are **very** unfortunate.

https://github.com/akhilrex/podgrab/blob/3179a875b8b638fb86d0e829d12a9761c1cd7f90/.env#L2
https://github.com/akhilrex/podgrab/blob/afd1696bfeb13949d18514f6138f98eaa144d292/db/db.go#L20

Maybe *something* like this would work:

```
systemd.services.podgrab = {
  environment = {
    DATA = "${pkgs.podgrab}/share/assets";
    CONFIG = "/var/lib/podgrab";
  };
  serviceConfig = {
    WorkingDirectory = "${pkgs.podgrab}/share";
  };
};

```

Please do what research you can to see how to avoid linking anything into this path (ie. the `client` directory).

Reply:
The problem with this is that `podgrab` wants to download files into `DATA`, which results in failures as it tries to create the files in the Nix store.

I'm not sure that I see a way to avoid this issue while also not linking `client` and `webassets` from the store.

Reply:
Sorry, I had confused what the `DATA` path represented.

I played around with this and was able to get it to work without symlinks. You should be able to do this:
```
systemd.services.podgrab = {
  environment = {
    DATA = "/var/lib/podgrab/data";
    CONFIG = "/var/lib/podgrab/config";
  };
  serviceConfig = {
    WorkingDirectory = "${pkgs.podgrab}/share";
    StateDirectory = [ "podgrab" "podgrab/config" "podgrab/data" ];
  };
};
```

Let me know if you have any issues.

![Screenshot_20210415_075154](https://user-images.githubusercontent.com/7755101/114864831-79a36d80-9dbf-11eb-9f40-9d46da8bfca6.png)

Reply:
That seems to work, thank you for checking it out :slightly_smiling_face: